Do not instrument classes loaded by platform class loader by default

For the following `src/Example.java`

```java
class Example {
    public static void main(String[] args) {
        new java.sql.SQLException();
        System.our.println("OK");
    }
}
```

compiled by JDK 8

```text
mkdir classes
javac src/Example.java -d classes
```

using JDK 8 execution of

```text
rm jacoco.exec
java -javaagent:jacoco-0.8.14/lib/jacocoagent.jar -cp classes Example
java -jar jacoco-0.8.14/lib/jacococli.jar execinfo jacoco.exec | grep SQLException | wc -l
```

produces 

```
OK
0
```

whereas execution using JDK 9 and above up to JDK 26 produces

```
OK
1
```

To instrument `java.sql.SQLException` prior to JDK 9 was required to use `inclbootstrapclasses=true`,
ie using JDK 8 execution of

```text
rm jacoco.exec
java -javaagent:jacoco-0.8.14/lib/jacocoagent.jar=inclbootstrapclasses=true -cp classes Example
java -jar jacoco-0.8.14/lib/jacococli.jar execinfo jacoco.exec | grep SQLException | wc -l
```

produces

```
OK
1
```

This difference comes from introduction of platform class loader by https://openjdk.org/jeps/261 in JDK 9.

By not instrumenting such classes by default
(eg by requiring for their instrumentation as it was before
`inclbootstrapclasses` to be `true` instead of default `false`)
we can get rid of exceptions when agent does not support bytecode version of classes in JDK EA
while supports bytecode version of classes whose coverage should be measured,
in which case such exceptions can already be safely ignored
(see https://github.com/jacoco/jacoco/issues/2151, https://github.com/jacoco/jacoco/issues/1742#issuecomment-2403330561 and https://github.com/jacoco/jacoco/pull/1631#issuecomment-2403535283) - ie execution using JDK 27 EA produces

```
java.lang.instrument.IllegalClassFormatException: Error while instrumenting java/sql/SQLException with JaCoCo 0.8.14.202510111229/2eb2483.
	at org.jacoco.agent.rt.internal_29a6edd.CoverageTransformer.transform(CoverageTransformer.java:94)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:257)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:594)
	at java.base/java.lang.ClassLoader.defineClass2(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1080)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1064)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:177)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:735)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassInModuleOrNull(BuiltinClassLoader.java:678)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:604)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:639)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:608)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:578)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:502)
	at Example.main(Example.java:3)
Caused by: java.io.IOException: Error while instrumenting java/sql/SQLException with JaCoCo 0.8.14.202510111229/2eb2483.
	at org.jacoco.agent.rt.internal_29a6edd.core.instr.Instrumenter.instrumentError(Instrumenter.java:161)
	at org.jacoco.agent.rt.internal_29a6edd.core.instr.Instrumenter.instrument(Instrumenter.java:111)
	at org.jacoco.agent.rt.internal_29a6edd.CoverageTransformer.transform(CoverageTransformer.java:92)
	... 15 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 71
	at org.jacoco.agent.rt.internal_29a6edd.asm.ClassReader.<init>(ClassReader.java:200)
	at org.jacoco.agent.rt.internal_29a6edd.asm.ClassReader.<init>(ClassReader.java:180)
	at org.jacoco.agent.rt.internal_29a6edd.asm.ClassReader.<init>(ClassReader.java:166)
	at org.jacoco.agent.rt.internal_29a6edd.core.internal.instr.InstrSupport.classReaderFor(InstrSupport.java:280)
	at org.jacoco.agent.rt.internal_29a6edd.core.instr.Instrumenter.instrument(Instrumenter.java:77)
	at org.jacoco.agent.rt.internal_29a6edd.core.instr.Instrumenter.instrument(Instrumenter.java:109)
	... 16 more
...
OK
0
```

while execution of

```
java -jar jacoco-0.8.14/lib/jacococli.jar report jacoco.exec \
    --classfiles classes \
    --sourcefiles src \
    --html report
```

still produces correct code coverage report for `Example.java`

And similarly for JaCoCo 0.8.15 and JDK 28 EA.

---

Should also be noted that such classes can already be excluded from instrumentation using
```
'excludes=java/sql/*'
```
or
```
'exclclassloader=jdk.internal.loader.ClassLoaders$PlatformClassLoader'
```

--- a/jacoco-maven-plugin/src/org/jacoco/maven/AbstractAgentMojo.java
+++ b/jacoco-maven-plugin/src/org/jacoco/maven/AbstractAgentMojo.java
@@ -88,9 +88,9 @@ public abstract class AbstractAgentMojo extends AbstractJacocoMojo {
 	@Parameter(property = "jacoco.exclClassLoaders")
 	String exclClassLoaders;
 	/**
-	 * Specifies whether also classes from the bootstrap classloader should be
-	 * instrumented. Use this feature with caution, it needs heavy
-	 * includes/excludes tuning.
+	 * Specifies whether also classes from the bootstrap and platform
+	 * classloaders should be instrumented. Use this feature with caution, it
+	 * needs heavy includes/excludes tuning.
 	 */
 	@Parameter(property = "jacoco.inclBootstrapClasses")
 	Boolean inclBootstrapClasses;

--- a/org.jacoco.agent.rt.test/src/org/jacoco/agent/rt/internal/CoverageTransformerTest.java
+++ b/org.jacoco.agent.rt.test/src/org/jacoco/agent/rt/internal/CoverageTransformerTest.java
@@ -14,6 +14,7 @@ package org.jacoco.agent.rt.internal;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
@@ -29,6 +30,7 @@ import org.jacoco.core.JaCoCo;
 import org.jacoco.core.internal.InputStreams;
 import org.jacoco.core.runtime.AbstractRuntime;
 import org.jacoco.core.runtime.AgentOptions;
+import org.jacoco.core.test.validation.JavaVersion;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
@@ -83,6 +85,42 @@ public class CoverageTransformerTest {
 		options.setInclBootstrapClasses(false);
 		CoverageTransformer t = createTransformer();
 		assertFalse(t.filter(null, "java/util/TreeSet", protectionDomain));
+	}
+
+	/**
+	 * Prior to introduction of platform class loader by JEP 261 in JDK 9
+	 * {@code java/sql/Timestamp} was loaded by bootstrap class loader - see
+	 * {@link #testFilterInclBootstrapClassesNegative()} and
+	 * {@link #platform_class_loader_should_be_available_starting_from_Java_9()}.
+	 *
+	 * @see #should_include_classes_loaded_by_platform_class_loader_when_inclbootstrap_classes_is_true()
+	 */
+	@Test
+	public void should_exclude_classes_loaded_by_platform_class_loader_when_inclbootstrap_classes_is_false() {
+		options.setInclBootstrapClasses(false);
+		CoverageTransformer t = createTransformer();
+		assertFalse(t.filter(CoverageTransformer.PLATFORM_LOADER,
+				"java/sql/Timestamp", protectionDomain));
+	}
+
+	/**
+	 * @see #should_exclude_classes_loaded_by_platform_class_loader_when_inclbootstrap_classes_is_false()
+	 */
+	@Test
+	public void should_include_classes_loaded_by_platform_class_loader_when_inclbootstrap_classes_is_true() {
+		options.setInclBootstrapClasses(true);
+		CoverageTransformer t = createTransformer();
+		assertTrue(t.filter(CoverageTransformer.PLATFORM_LOADER,
+				"java/sql/Timestamp", protectionDomain));
+	}
+
+	@Test
+	public void platform_class_loader_should_be_available_starting_from_Java_9() {
+		if (JavaVersion.current().isBefore("9")) {
+			assertNull(CoverageTransformer.PLATFORM_LOADER);
+		} else {
+			assertNotNull(CoverageTransformer.PLATFORM_LOADER);
+		}
 	}
 
 	@Test

--- a/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
+++ b/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
@@ -146,10 +146,11 @@ public class CoverageTransformer implements ClassFileTransformer {
 	 * {@code java.lang.ClassLoader::getPlatformClassLoader}</a> introduced in
 	 * JDK 9 by <a href="https://openjdk.org/jeps/261">JEP 261</a>.
 	 *
-	 * @return platform class loader, or {@code null} if method for retrieving
-	 *         it is unavailable
+	 * @return platform class loader, or {@code null} if no such method
 	 * @throws IllegalStateException
-	 *             if exception occurs while retrieving platform class loader
+	 *             if {@link IllegalAccessException} or
+	 *             {@link InvocationTargetException} occurs during invocation
+	 * @see #PLATFORM_LOADER
 	 */
 	private static ClassLoader getPlatformClassLoader() {
 		try {

--- a/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
+++ b/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
@@ -14,6 +14,7 @@ package org.jacoco.agent.rt.internal;
 
 import java.lang.instrument.ClassFileTransformer;
 import java.lang.instrument.IllegalClassFormatException;
+import java.lang.reflect.InvocationTargetException;
 import java.security.CodeSource;
 import java.security.ProtectionDomain;
 
@@ -26,6 +27,11 @@ import org.jacoco.core.runtime.WildcardMatcher;
  * Class file transformer to instrument classes for code coverage analysis.
  */
 public class CoverageTransformer implements ClassFileTransformer {
+
+	/**
+	 * Cached result of {@link #getPlatformClassLoader()}.
+	 */
+	static final ClassLoader PLATFORM_LOADER = getPlatformClassLoader();
 
 	private static final String AGENT_PREFIX;
 
@@ -132,6 +138,31 @@ public class CoverageTransformer implements ClassFileTransformer {
 				includes.matches(classname) &&
 
 				!excludes.matches(classname);
+	}
+
+	/**
+	 * Invokes <a href=
+	 * "https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassLoader.html#getPlatformClassLoader()">
+	 * {@code java.lang.ClassLoader::getPlatformClassLoader}</a> introduced in
+	 * JDK 9 by <a href="https://openjdk.org/jeps/261">JEP 261</a>.
+	 *
+	 * @return platform class loader, or {@code null} if method for retrieving
+	 *         it is unavailable
+	 * @throws IllegalStateException
+	 *             if exception occurs while retrieving platform class loader
+	 */
+	private static ClassLoader getPlatformClassLoader() {
+		try {
+			return (ClassLoader) ClassLoader.class
+					.getMethod("getPlatformClassLoader").invoke(null);
+		} catch (final NoSuchMethodException e) {
+			// prior to JDK 9
+			return null;
+		} catch (final IllegalAccessException e) {
+			throw new IllegalStateException(e);
+		} catch (final InvocationTargetException e) {
+			throw new IllegalStateException(e);
+		}
 	}
 
 	/**

--- a/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
+++ b/org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/CoverageTransformer.java
@@ -119,7 +119,7 @@ public class CoverageTransformer implements ClassFileTransformer {
 	 */
 	boolean filter(final ClassLoader loader, final String classname,
 			final ProtectionDomain protectionDomain) {
-		if (loader == null) {
+		if (loader == null || loader == PLATFORM_LOADER) {
 			if (!inclBootstrapClasses) {
 				return false;
 			}

--- a/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
@@ -158,6 +158,23 @@
 		<au:assertLogContains text="java/sql/Timestamp"/>
 	</target>
 
+	<target name="testExclBootstrapClasses">
+		<jacoco:coverage destfile="${exec.file}">
+			<java classname="org.jacoco.ant.TestTarget" fork="true" failonerror="true">
+				<classpath path="${org.jacoco.ant.coverageTaskTest.classes.dir}"/>
+			</java>
+		</jacoco:coverage>
+
+		<au:assertLogContains text="Enhancing java with coverage"/>
+		<au:assertFileExists file="${exec.file}"/>
+		<au:assertLogContains text="Target executed"/>
+
+		<java classname="org.jacoco.ant.DumpExecClassNames" fork="true" classpath="${java.class.path}" failonerror="true">
+			<arg value="${exec.file}" />
+		</java>
+		<au:assertLogDoesntContain text="java/sql/Timestamp"/>
+	</target>
+
 	<target name="testIllegalReflectiveAccess">
 		<jacoco:coverage destfile="${exec.file}">
 			<java classname="org.jacoco.ant.IllegalReflectiveAccessTarget" fork="true" failonerror="true">

--- a/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
@@ -142,7 +142,7 @@
 	</target>
 
 	<target name="testInclBootstrapClasses">
-		<jacoco:coverage destfile="${exec.file}" inclbootstrapclasses="true" includes="java/sql/*">
+		<jacoco:coverage destfile="${exec.file}" inclbootstrapclasses="true" includes="java/util/*:java/sql/*">
 			<java classname="org.jacoco.ant.TestTarget" fork="true" failonerror="true">
 				<classpath path="${org.jacoco.ant.coverageTaskTest.classes.dir}"/>
 			</java>
@@ -155,11 +155,14 @@
 		<java classname="org.jacoco.ant.DumpExecClassNames" fork="true" classpath="${java.class.path}" failonerror="true">
 			<arg value="${exec.file}" />
 		</java>
+		<!-- class from bootstrap classloader -->
+		<au:assertLogContains text="java/util/UUID"/>
+		<!-- class from platform classloader introduced by JEP 261 in JDK 9, formerly from bootstrap classloader -->
 		<au:assertLogContains text="java/sql/Timestamp"/>
 	</target>
 
 	<target name="testExclBootstrapClasses">
-		<jacoco:coverage destfile="${exec.file}" inclbootstrapclasses="false" includes="java/sql/*">
+		<jacoco:coverage destfile="${exec.file}" inclbootstrapclasses="false" includes="java/util/*:java/sql/*">
 			<java classname="org.jacoco.ant.TestTarget" fork="true" failonerror="true">
 				<classpath path="${org.jacoco.ant.coverageTaskTest.classes.dir}"/>
 			</java>
@@ -172,6 +175,9 @@
 		<java classname="org.jacoco.ant.DumpExecClassNames" fork="true" classpath="${java.class.path}" failonerror="true">
 			<arg value="${exec.file}" />
 		</java>
+		<!-- class from bootstrap classloader -->
+		<au:assertLogDoesntContain text="java/util/UUID"/>
+		<!-- class from platform classloader introduced by JEP 261 in JDK 9, formerly from bootstrap classloader -->
 		<au:assertLogDoesntContain text="java/sql/Timestamp"/>
 	</target>
 

--- a/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/CoverageTaskTest.xml
@@ -159,7 +159,7 @@
 	</target>
 
 	<target name="testExclBootstrapClasses">
-		<jacoco:coverage destfile="${exec.file}">
+		<jacoco:coverage destfile="${exec.file}" inclbootstrapclasses="false" includes="java/sql/*">
 			<java classname="org.jacoco.ant.TestTarget" fork="true" failonerror="true">
 				<classpath path="${org.jacoco.ant.coverageTaskTest.classes.dir}"/>
 			</java>

--- a/org.jacoco.ant.test/src/org/jacoco/ant/InstrumentTaskTest.xml
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/InstrumentTaskTest.xml
@@ -63,7 +63,7 @@
 		<jacoco:instrument destdir="${instr.dir}">
 			<fileset dir="${lib.dir}" includes="*.jar"/>
 		</jacoco:instrument>
-		<au:assertLogContains text="Instrumented 20 classes to ${temp.dir}"/>
+		<au:assertLogContains text="Instrumented 21 classes to ${temp.dir}"/>
 
 		<unzip src="${instr.dir}/test.jar" dest="${instr.dir}"/>
 		<au:assertFileDoesntExist file="${instr.dir}/META-INF/TEST.RSA" />
@@ -84,7 +84,7 @@
 		<jacoco:instrument destdir="${instr.dir}" removesignatures="false">
 			<fileset dir="${lib.dir}" includes="*.jar"/>
 		</jacoco:instrument>
-		<au:assertLogContains text="Instrumented 20 classes to ${temp.dir}"/>
+		<au:assertLogContains text="Instrumented 21 classes to ${temp.dir}"/>
 
 		<unzip src="${instr.dir}/test.jar" dest="${instr.dir}"/>
 		<au:assertFileExists file="${instr.dir}/META-INF/TEST.RSA" />
@@ -95,7 +95,7 @@
 		<jacoco:instrument destdir="${temp.dir}">
 			<fileset dir="${org.jacoco.ant.instrumentTaskTest.classes.dir}" includes="**/*.class"/>
 		</jacoco:instrument>
-		<au:assertLogContains text="Instrumented 20 classes to ${temp.dir}"/>
+		<au:assertLogContains text="Instrumented 21 classes to ${temp.dir}"/>
 		<au:assertFileExists file="${temp.dir}/org/jacoco/ant/InstrumentTaskTest.class" />
 
 		<echo file="${temp.dir}/jacoco-agent.properties">destfile=test.exec</echo>
@@ -112,7 +112,7 @@
 		<jacoco:instrument destdir="${temp.dir}">
 			<fileset dir="${org.jacoco.ant.instrumentTaskTest.classes.dir}" includes="**/*.class"/>
 		</jacoco:instrument>
-		<au:assertLogContains text="Instrumented 20 classes to ${temp.dir}"/>
+		<au:assertLogContains text="Instrumented 21 classes to ${temp.dir}"/>
 		<au:assertFileExists file="${temp.dir}/org/jacoco/ant/InstrumentTaskTest.class" />
 
 		<java classname="org.jacoco.ant.TestTarget" failonerror="true" fork="true">
@@ -129,7 +129,7 @@
 		<jacoco:instrument destdir="${temp.dir}">
 			<fileset dir="${org.jacoco.ant.instrumentTaskTest.classes.dir}" includes="**/*.class"/>
 		</jacoco:instrument>
-		<au:assertLogContains text="Instrumented 20 classes to ${temp.dir}"/>
+		<au:assertLogContains text="Instrumented 21 classes to ${temp.dir}"/>
 		<au:assertFileExists file="${temp.dir}/org/jacoco/ant/InstrumentTaskTest.class" />
 
 		<java classname="org.jacoco.ant.TestTarget" failonerror="false" fork="true">

--- a/org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTarget.java
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTarget.java
@@ -1,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2026 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.ant;
+
+/**
+ * Test target for {@link SecurityManagerTest} that does not require any
+ * permissions. To assert execution it prints "Target executed".
+ */
+public class SecurityManagerTarget {
+
+	public static void main(String[] args) throws Exception {
+		System.out.println("Target executed");
+	}
+
+}

--- a/org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.xml
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.xml
@@ -26,7 +26,7 @@
 
 	<target name="testJaCoCoWithSecurityManager">
 		<jacoco:coverage destfile="${exec.file}">
-			<java classname="org.jacoco.ant.TestTarget" fork="true" failonerror="true">
+			<java classname="org.jacoco.ant.SecurityManagerTarget" fork="true" failonerror="true">
 				<classpath path="${org.jacoco.ant.securityManagerTest.classes.dir}"/>
 				<jvmarg value="-Djacoco.agent=${_jacoco.agentFile}"/>
 				<jvmarg value="-Djacoco.exec=${exec.file}"/>

--- a/org.jacoco.ant.test/src/org/jacoco/ant/TestTarget.java
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/TestTarget.java
@@ -32,8 +32,20 @@ public class TestTarget {
 
 	public static void main(String[] args) throws Exception {
 
-		// Load some class from the bootstrap classloader:
+		// Load some class from the bootstrap classloader
+		new java.util.UUID(0, 0);
+		if (java.util.UUID.class.getClassLoader() != null) {
+			throw new AssertionError();
+		}
+		// and from the platform classloader
+		// introduced by JEP 261 in JDK 9
 		new java.sql.Timestamp(0);
+		if (java.sql.Timestamp.class.getClassLoader() != null
+				&& !"jdk.internal.loader.ClassLoaders$PlatformClassLoader"
+						.equals(java.sql.Timestamp.class.getClassLoader()
+								.getClass().getName())) {
+			throw new AssertionError();
+		}
 
 		System.out.println("Target executed");
 

--- a/org.jacoco.ant.test/src/org/jacoco/ant/data/policy.txt
+++ b/org.jacoco.ant.test/src/org/jacoco/ant/data/policy.txt
@@ -6,4 +6,5 @@ grant codeBase "file:${jacoco.agent}" {
     permission java.lang.RuntimePermission "getProtectionDomain";
     permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
     permission java.lang.RuntimePermission "defineClass";
+    permission java.lang.RuntimePermission "getClassLoader";
 };

--- a/org.jacoco.ant/src/org/jacoco/ant/AbstractCoverageTask.java
+++ b/org.jacoco.ant/src/org/jacoco/ant/AbstractCoverageTask.java
@@ -116,11 +116,12 @@ public class AbstractCoverageTask extends Task {
 	}
 
 	/**
-	 * Sets whether classes from the bootstrap classloader should be
-	 * instrumented.
+	 * Sets whether classes from the bootstrap and platform classloaders should
+	 * be instrumented.
 	 *
 	 * @param include
-	 *            <code>true</code> if bootstrap classes should be instrumented
+	 *            <code>true</code> if classes from bootstrap and platform
+	 *            classloaders should be instrumented
 	 */
 	public void setInclBootstrapClasses(final boolean include) {
 		agentOptions.setInclBootstrapClasses(include);

--- a/org.jacoco.core/src/org/jacoco/core/runtime/AgentOptions.java
+++ b/org.jacoco.core/src/org/jacoco/core/runtime/AgentOptions.java
@@ -79,9 +79,9 @@ public final class AgentOptions {
 	public static final String EXCLCLASSLOADER = "exclclassloader";
 
 	/**
-	 * Specifies whether also classes from the bootstrap classloader should be
-	 * instrumented. Use this feature with caution, it needs heavy
-	 * includes/excludes tuning. Default is <code>false</code>.
+	 * Specifies whether also classes from the bootstrap and platform
+	 * classloaders should be instrumented. Use this feature with caution, it
+	 * needs heavy includes/excludes tuning. Default is <code>false</code>.
 	 */
 	public static final String INCLBOOTSTRAPCLASSES = "inclbootstrapclasses";
 
@@ -361,22 +361,23 @@ public final class AgentOptions {
 	}
 
 	/**
-	 * Returns whether classes from the bootstrap classloader should be
-	 * instrumented.
+	 * Returns whether classes from the bootstrap and platform classloaders
+	 * should be instrumented.
 	 *
-	 * @return <code>true</code> if classes from the bootstrap classloader
-	 *         should be instrumented
+	 * @return <code>true</code> if classes from the bootstrap and platform
+	 *         classloaders should be instrumented
 	 */
 	public boolean getInclBootstrapClasses() {
 		return getOption(INCLBOOTSTRAPCLASSES, false);
 	}
 
 	/**
-	 * Sets whether classes from the bootstrap classloader should be
-	 * instrumented.
+	 * Sets whether classes from the bootstrap and platform classloaders should
+	 * be instrumented.
 	 *
 	 * @param include
-	 *            <code>true</code> if bootstrap classes should be instrumented
+	 *            <code>true</code> if classes from the bootstrap and platform
+	 *            classloaders should be instrumented
 	 */
 	public void setInclBootstrapClasses(final boolean include) {
 		setOption(INCLBOOTSTRAPCLASSES, include);

--- a/org.jacoco.doc/docroot/doc/agent.html
+++ b/org.jacoco.doc/docroot/doc/agent.html
@@ -120,9 +120,9 @@
     </tr>
     <tr>
       <td><code>inclbootstrapclasses</code></td>
-      <td>Specifies whether also classes from the bootstrap classloader should
-          be instrumented. Use this feature with caution, it needs heavy
-          includes/excludes tuning.
+      <td>Specifies whether also classes from the bootstrap and platform
+          classloaders should be instrumented. Use this feature with caution,
+          it needs heavy includes/excludes tuning.
       </td>
       <td><code>false</code></td>
     </tr>

--- a/org.jacoco.doc/docroot/doc/ant.html
+++ b/org.jacoco.doc/docroot/doc/ant.html
@@ -210,9 +210,9 @@
     </tr>
     <tr>
       <td><code>inclbootstrapclasses</code></td>
-      <td>Specifies whether also classes from the bootstrap classloader should
-          be instrumented. Use this feature with caution, it needs heavy
-          includes/excludes tuning.
+      <td>Specifies whether also classes from the bootstrap and platform
+          classloaders should be instrumented. Use this feature with caution,
+          it needs heavy includes/excludes tuning.
       </td>
       <td><code>false</code></td>
     </tr>

--- a/org.jacoco.doc/docroot/doc/changes.html
+++ b/org.jacoco.doc/docroot/doc/changes.html
@@ -20,6 +20,13 @@
 
 <h2>Snapshot Build @qualified.bundle.version@ (@build.date@)</h2>
 
+<h3>New Features</h3>
+<ul>
+  <li>To instrument classes loaded by the platform class loader, introduced in JDK 9
+      by JEP 261, the agent option <code>inclbootstrapclasses=true</code> is required
+      (GitHub <a href="https://github.com/jacoco/jacoco/issues/2155">#2155</a>).</li>
+</ul>
+
 <h2>Release 0.8.15 (2026/06/04)</h2>
 
 <h3>New Features</h3>

